Mixpanel/Mixpanel.m

```diff
@@ -606,10 +616,40 @@ - (void)flushQueue:(NSMutableArray *)queue endpoint:(NSString *)endpoint
         NSString *postBody = [NSString stringWithFormat:@"ip=%d&data=%@", self.useIPAddressForGeoLocation, requestData];
         MixpanelDebug(@"%@ flushing %lu of %lu to %@: %@", self, (unsigned long)batch.count, (unsigned long)queue.count, endpoint, queue);
         NSURLRequest *request = [self apiRequestWithEndpoint:endpoint andBody:postBody];
-        NSError *error = nil;
 
         [self updateNetworkActivityIndicator:YES];
         
+#if defined(MIXPANEL_TVOS_EXTENSION)
+        __block BOOL didFail = NO;
+        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+        NSURLSession *session = [NSURLSession sharedSession];
+        [[session dataTaskWithRequest:request completionHandler:^(NSData *responseData,
+                                                                  NSURLResponse *urlResponse,
+                                                                  NSError *error) {
+            [self updateNetworkActivityIndicator:NO];
+            
+            BOOL success = [self handleNetworkResponse:(NSHTTPURLResponse *)urlResponse withError:error];
+            if (error || !success) {
+                MixpanelError(@"%@ network failure: %@", self, error);
+                didFail = YES;
+            }
+            else {
+                NSString *response = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
+                if ([response intValue] == 0) {
+                    MixpanelError(@"%@ %@ api rejected some items", self, endpoint);
+                }
+            }
+            
+            dispatch_semaphore_signal(semaphore);
+        }] resume];
+        
+        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER);
+        
+        if (didFail) {
+            break;
+        }
+#else
+        NSError *error = nil;
```

Comment:
Nit: Would like to see the duplication removed here. But since we're about to refactor this, don't worry about it.